apx/server/src/server_main.c

```diff
@@ -77,13 +77,22 @@ int main(int argc, char **argv)
    apx_server_create(&m_server,m_port);
    apx_server_setDebugMode(&m_server, g_debug);
    apx_server_start(&m_server);
+
+   signal(SIGINT, main_signal_handler);
+   signal(SIGTERM, main_signal_handler);
    for(;;)
    {
-      SLEEP(5000); //main thread is sleeping while child threads do all the work
-/*    if (++m_count==20) //this counter is used during testing to verify that all resources are properly cleaned up
+      //main thread is sleeping while child threads do all the work
+#ifdef _WIN32
+      SLEEP(5000);
+      if(m_running == 0)
       {
          break;
-      }*/
+      }
+#else
+      pause();

```

Comment:
There is pause function linux? cool :-)

Reply:
Yea. Worked in cygwin (win32 is not set) as well. Unpause occurs when a signal with a handler is sent to the program, so no need for sleep loop.

In cygwin on w10 the sleep however seems to be aborted on sigint, so no extra latency there either...